File: typescript/examples/src/storing.ts
import {
  HuggingFaceEmbedding,
  HuggingFaceEmbeddingModelType,
  serviceContextFromDefaults,
  SimpleDirectoryReader,
  storageContextFromDefaults,
  VectorStoreIndex,
} from 'llamaindex';
import { CHUNK_OVERLAP, CHUNK_SIZE, STORAGE_CACHE_DIR, STORAGE_DIR } from './utils/constants';

const documents = await new SimpleDirectoryReader().loadData(STORAGE_DIR);

console.log(`Loaded ${documents.length} documents.`);

const storageContext = await storageContextFromDefaults({
  persistDir: STORAGE_CACHE_DIR,
});

const embedModel = new HuggingFaceEmbedding({
  modelType: HuggingFaceEmbeddingModelType.XENOVA_ALL_MINILM_L6_V2,
});

const serviceContext = serviceContextFromDefaults({
  chunkSize: CHUNK_SIZE,
  chunkOverlap: CHUNK_OVERLAP,
  embedModel,
  llm: 1 as any, // unused
});

console.log(`Generating storage context...`);

await VectorStoreIndex.fromDocuments(documents, {
  storageContext,
  serviceContext,
});

console.log('Storage context successfully generated.');

File: python/example/src/storing.py
import os

from llama_index import SimpleDirectoryReader, StorageContext, ServiceContext, VectorStoreIndex
from llama_index.embeddings import HuggingFaceEmbedding

from utils.constants import STORAGE_DIR, STORAGE_CACHE_DIR, CHUNK_SIZE, CHUNK_OVERLAP

reader = SimpleDirectoryReader(input_dir=STORAGE_DIR)

documents = reader.load_data()

print(f"Loaded {len(documents)} documents.")

embedModel = HuggingFaceEmbedding(model_name="BAAI/bge-small-en-v1.5")

service_context = ServiceContext.from_defaults(
    chunk_size=CHUNK_SIZE,
    chunk_overlap=CHUNK_OVERLAP,
    embed_model=embedModel,
    llm=None,
)

print('Generating storage context...')

if os.path.exists(STORAGE_CACHE_DIR):
    storage_context = StorageContext.from_defaults(
        persist_dir=STORAGE_CACHE_DIR
    )

    index = VectorStoreIndex.from_documents(
        documents=documents,
        storage_context=storage_context,
        service_context=service_context
    )
else:
    index = VectorStoreIndex.from_documents(
        documents=documents,
        service_context=service_context
    )

    index.storage_context.persist(persist_dir=STORAGE_CACHE_DIR)

print('Storage context successfully generated.');